internal/rbd/rbd_journal.go

```diff
@@ -652,6 +664,27 @@ func RegenerateJournal(
 	return rbdVol.VolID, nil
 }
 
+// setVolumeDetailsAsMetadata set volume details as Image metadata.
+func (rv *rbdVolume) setVolumeDetailsAsMetadata(
+	volumeAttributes map[string]string,
+	claimName,
+	claimNamespace string,
+	setMetadata bool) error {
+	if setMetadata &&
+		claimName != volumeAttributes[pvcNameKey] || claimNamespace != volumeAttributes[pvcNamespaceKey] {
```

Comment:
This check is not required as we don't store the PVC name and namespace name in the volume attributes, I think you also need to set the PV name also as once can create a  new PV for the existing rbd image and use it for the new PVC.

Reply:
We don't store PVC name and namespace name in the volume attributes, with your recent changes?

Reply:
Yes correct